src/Legend/Legend.tsx

```diff
@@ -139,28 +144,30 @@ export const Legend: React.FC<LegendProps> = ({
       className={`${CSS_PREFIX}legend ${className ? className : ''}`}
       {...passThroughProps}
     >
-      <Spin
-        spinning={loading}
-        indicator={(
-          <FontAwesomeIcon
-            icon={faCircleNotch}
-          />
-        )}
-      >
-        {legendError ?
-          <div
-            className='legend-load-error'
-          >
-            {errorMsg}
-          </div>
-          :
-          <img
-            src={imgSrc}
-            alt={alt}
-            onError={handleError.bind(this)}
-          />
-        }
-      </Spin>
+      { loadingMask &&
+        <Spin
+          spinning={loading}
+          indicator={(
+            <FontAwesomeIcon
+              icon={faCircleNotch}
+            />
+          )}
+        >
+          {legendError ?
+            <div
+              className='legend-load-error'
+            >
+              {errorMsg}
+            </div>
+            :
+            <img
+              src={imgSrc}
+              alt={alt}
+              onError={handleError.bind(this)}
+            />
+          }
+        </Spin>
+      }
```

Comment:
If the loading mask is not visible, the whole legend won't be visible. I don't think this is what you want?

Reply:
You are absolutely correct! That should be fixed now